shapes/main.cpp

```diff
@@ -1,80 +1,90 @@
-#include <iostream>
-#include <vector>
 #include <algorithm>
-#include <string>
+#include <functional>
+#include <iostream>
+#include <iterator>
+#include <map>
 #include <memory>
-#include "Shape.hpp"
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include "Circle.hpp"
 #include "Rectangle.hpp"
+#include "Shape.hpp"
 #include "Square.hpp"
-#include "Circle.hpp"
 
 using namespace std;
 
 using Collection = vector<shared_ptr<Shape>>;
 
-bool sortByArea(shared_ptr<Shape> first, shared_ptr<Shape> second)
+//TASK #11
+template <class DerivedType>
+using SFINAE = std::enable_if<std::is_base_of<Circle, std::remove_reference_t<DerivedType>>::value>;
```

Comment:
Hmm... Circle? We want to enable this overload for all classes that inherit from Shape, not Circle.
i'm not sure you should also take a type of this whole expression
```suggestion
using SFINAE = std::enable_if<std::is_base_of<Shape, std::remove_reference_t<DerivedType>>::value>::type;
```